src/Components/WebAssembly/WebAssembly/src/Rendering/WebAssemblyRenderer.cs

```diff
@@ -101,18 +106,7 @@ protected override Task UpdateDisplayAsync(in RenderBatch batch)
         /// <inheritdoc />
         protected override void HandleException(Exception exception)
         {
-            Console.Error.WriteLine($"Unhandled exception rendering component:");
-            if (exception is AggregateException aggregateException)
-            {
-                foreach (var innerException in aggregateException.Flatten().InnerExceptions)
-                {
-                    Console.Error.WriteLine(innerException);
-                }
-            }
-            else
-            {
-                Console.Error.WriteLine(exception);
-            }
+            _logger.LogCritical(exception, "Unhandled exception");
```

Comment:
Any reason this isn't the same pattern as the server: https://github.com/dotnet/aspnetcore/blob/master/src/Components/Server/src/Circuits/RemoteRenderer.cs#L388-L391?

Reply:
Good point.

I'm changing `WebAssemblyConsoleLogger` and `WebAssemblyLoggerFactory` to be as similar to the server-side implementations as I think makes sense. There are still some deliberate differences:

 * In Blazor Server, the severity of an unhandled component rendering exception is "warning", because it's not fatal to the server (only to the circuit). In Blazor WebAssembly, the severity for this is "critical" because it marks the point where the whole environment enters an undefined state and we show the gold "error occurred" bar. It makes sense for these to be different because the logging is done from the point of view of a bigger environment on the server.
 * The exact details of message formatting are *mostly* the same, but not precisely. This is because the browser logger (1) doesn't want to see trailing newlines and (2) has its own built-in behavior around colorization and highlighting different severity levels.
 * The WebAssembly implementations still don't support configuring log levels, because (1) the configuration feature isn't available yet, and (2) it's out of scope for this PR, and tracked by https://github.com/dotnet/aspnetcore/issues/19737. So it's still hardcoded to "warning" level and above.